snouter/src/auth/product-owner.guard.ts

```diff
@@ -0,0 +1,22 @@
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+import { ProductsService } from 'src/products/products.service';
+
+@Injectable()
+export class ProductOwnerGuard implements CanActivate {
+  constructor(private readonly productService: ProductsService) {}
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const request = context.switchToHttp().getRequest();
+    const user = request.user;
+    const productId = request.params.id;
+    const productsForUser = await this.productService.getProductsForUser(
+      user.id,
+    );
+
+    const isSelling = [...productsForUser].find(
+      (product) => +product.id === +productId,
+    );
```

Comment:
this part of code seems kinda sus
- no need to cast `product.id` to integer since it already is
- no need to cast `productId` to integer every time (you can do when assigning variable on line 11)
- no need for destructuring array since `.find` does not change any data
- since `isSelling` hints that it's a boolean but it really is product or null, I'd rather move `!!` here